Migrate from deprecated pedantic to flutter_lints

adityathakurxd opened the pull request:
## Description

Migrated all packages from deprecated `pedantic` to `flutter_lints`. 
Added dev_dependency on package: `flutter_lints` to all project’s pubspec.yaml

Since the project already had a custom `analysis_options.yaml` file at its root, I added `include: package:flutter_lints/flutter.yaml` to it.


## Related Issues

#471

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

miquelbeltran commented:
Change looks good to me, need to finish the static analysis and see if it all works, then it can be approved and merged

miquelbeltran commented:
As you can see @adityathakurxd there's a bunch of static analysis issues that came out since the `flutter_lints` is introduced, you would have to fix those before merging: https://github.com/fluttercommunity/plus_plugins/pull/519/checks?check_run_id=3811595912

adityathakurxd commented:
Yes, I'll look into this @miquelbeltran Thank you for reviewing.

I wanted to ask like some of these messages in https://github.com/fluttercommunity/plus_plugins/pull/519/checks?check_run_id=3811595912 include `Avoid print calls in production code` but these statements are used to `print(e.toString());` or other such cases. I wanted to ask it would be right removing so should the fix be in analysis_option to allow print in production?

miquelbeltran commented:
I am OK with plugins using `print()` in case of errors, so we can add the rule to the `analysis_options.yaml` to ignore it.

I would also considering having a global `analysis_options.yaml` where we can define all the rules at the root of the project and linking to that, if that's possible!

adityathakurxd commented:
Okay. I'll add the rule. There is a global `analysis_options.yaml` at the root. I'll work with that.

vbuberen commented:
@miquelbeltran I would suggest switching from `print` to `dart:developer` as it is suggested here https://flutter.dev/docs/testing/code-debugging#logging
Or there is a specific reason to stick with `print`?

miquelbeltran commented:
> @miquelbeltran I would suggest switching from `print` to `dart:developer` as it is suggested here https://flutter.dev/docs/testing/code-debugging#logging Or there is a specific reason to stick with `print`?

love the idea! I would suggest to do it on a separated PR after this one if possible.

miquelbeltran commented:
actually no need to wait for this PR ;)

adityathakurxd commented:
I have updated the PR with the required changes @miquelbeltran 
For the print in production issue, to the global `analysis_options.yaml` I added:
```
 errors:
    avoid_print: ignore
```
and, the `melos run analyze` is running fine now.
![Administrator_ Command Prompt - melos  run analyze 10_6_2021 4_09_19 PM (2)](https://user-images.githubusercontent.com/53579386/136188972-aa60e20b-4372-4c1f-9f62-1b1d08a129f5.png)

miquelbeltran commented:
thank you @adityathakurxd seems that the analyze is passing I will merge once CI finishes

adityathakurxd commented:
Thank you so much @miquelbeltran 🎉
